### PR TITLE
Lead federation school homepage

### PR DESCRIPTION
**Pull request by J05h-L**
### Context
[AB#232745](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/232745) - [AB#233273](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/233273)

### Change proposed in this pull request
Adds a section to the homepage to be displayed when said school is a lead federation school. This contains a brief message that the school is the lead school and a list of school within that federation.
Minor change to platform to return the federated schools in the response.
Minor change to update the existing integration tests for this page.

### Guidance to review 
Integration tests to follow in a future PR

### Checklist (add/remove as appropriate)
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally
- [x] You have reviewed with UX/Design

### CODE REVIEW

File: platform/src/apis/Platform.Api.Establishment/Schools/SchoolsService.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public class SchoolsService(ISearchConnection<School> searchConnection, IDatabas
         using var conn = await dbFactory.GetConnection();
         var school = await conn.QueryFirstOrDefaultAsync<School>(schoolSql, new { URN = urn });
 
-        if (school != null && string.IsNullOrEmpty(school.FederationLeadURN))
+        if (school != null && !string.IsNullOrEmpty(school.FederationLeadURN))
         {
             school.Schools = await conn.QueryAsync<School>(childSchoolsSql, new { URN = urn });
```
Comment (WolfyUK) [inline platform/src/apis/Platform.Api.Establishment/Schools/SchoolsService.cs:31]: chore: To be explicit, please can this property be renamed FederatedSchools (or similar)

File: web/src/Web.App/Views/School/Index.cshtml [hunk 2]
```diff
@@ -39,6 +39,11 @@
     @await Html.PartialAsync("_Banners", Model)
 </div>
 
+@if (Model.FederationSchools.Length > 0)
```
Comment (WolfyUK) [inline web/src/Web.App/Views/School/Index.cshtml:42]: query: There are 30 schools in `d01` that are alone within a federation. Should this, or an alternative partial be rendered in this case? 
```
100472
101264
101266
101290
101294
101295
101299
108426
111003
114053
114063
114064
114065
114311
114340
116715
116716
116774
116960
118318
118471
121500
121565
122641
122680
122766
133397
134171
134662
134663
```
A related question is, why are these alone?

File: platform/src/apis/Platform.Api.Establishment/Schools/SchoolsService.cs (hunk 1 above)
Comment (WolfyUK) [reply platform/src/apis/Platform.Api.Establishment/Schools/SchoolsService.cs:31]: It looks like this was added [here](https://github.com/DFE-Digital/education-benchmarking-and-insights/pull/1476/files#diff-3408b98052ad41104923ccc5753eb61bd369cb15278fea131979a105d9af43c5)

File: web/src/Web.App/Views/School/Index.cshtml (hunk 2 above)
Comment (jrabbott) [reply web/src/Web.App/Views/School/Index.cshtml:42]: The logic in the front-end for me is correct and we shouldn't be adding logic to correct for the data.

This data issue is caused by the CFR data. In 2023 for some reason there are no non-lead schools in the submission.

File: platform/src/apis/Platform.Api.Establishment/Schools/SchoolsService.cs (hunk 1 above)
Comment (jrabbott) [reply platform/src/apis/Platform.Api.Establishment/Schools/SchoolsService.cs:31]: Agreed 👍

**Review by WolfyUK — CHANGES_REQUESTED** [review]
Apart from the property rename, this looks great and works well!

File: web/src/Web.App/Views/School/Index.cshtml (hunk 2 above)
Comment (jrabbott) [reply web/src/Web.App/Views/School/Index.cshtml:42]: As a result of the work we've done on the CFR process we'll be better informed on the submission data.

**Review by WolfyUK — APPROVED** [review]

**Pull request merged**